lib/util/timeout.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+/* eslint consistent-return:0 */
+function TimeoutHandler () {
+	this.disable = true;
+}
+
+TimeoutHandler.prototype.setOptions = function ( options ) {
+	if ( options ) {
+		this.disable = false;
+		var defaultConfig = {
+			'ms' : 40000
+		};
+		this.options = options || defaultConfig;
+	}
+};
+
+TimeoutHandler.prototype.set = function ( options, cb ) {
+
+	if ( this.disable ) {
+		return;
+	}
+
+	var respond = options.respond;
+	var log     = options.log;
+	var emitter = options.emitter;
+
+	return setTimeout( function () {
+		respond( {
+			'status' : 'fail',
+			'data'   : 'Request timeout'
+		} );
+		emitter.emit( 'timeout', {
+			'messageType' : options.messageType,
+			'message'     : options.message
+		} );
+		setImmediate( function () {
+			log.warn( 'timeout', {
```

Comment:
Should this be a warn or error? It seems like if this happens we have bigger problems.